spec/type_references_and_ownership.md

```diff
@@ -0,0 +1,503 @@
+group/kind information all names they supply must point to the right kind of resource. If they are using the `AzureReference` portion
+of `AugmentedKnownTypeReference` we can look up the expected format of the ARM ID for the resource type corresponding to the referenced GVK
+and validate that the customer has provided the correct one.
+
+### Open Design Question - shape of Azure References
+Instead of having an `AzureReference` as proposed above, we could instead just have the Kubernetes shaped
+reference, and a way where customers of the operator could import external Azure resources into Kubernetes in
+an unmanaged mode. Resources in this unmanaged mode couldn't be updated/modified through Kubernetes, and 
+if deleted it would just delete the k8s representation of the object, not the actual backing object in ARM.
+
+This is what [CAPZ does](https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/d93e75deff2be3c7647c2ddfd387ce0ef022e1cd/api/v1alpha2/tags.go#L78) today.
+
+#### Advantages compared the proposal above
+- References always look very Kubernetes native. This has an advantage especially when a customer is referring to a single
+  shared resource (Storage Account or the like) across many YAMLs. The more usages they have of it the more awkward it
+  gets to refer to it by fully qualified ARM ID all the time.
+- The structure of the reference type would be simpler because it's not `Azure` or `Kubernetes`, it could just look like
+  `KnownTypeReference` does today in k8s-infra.
+
+#### Disadvantages compared to the proposal above
+- It means that you can only reference other resources which are being tracked by Kubernetes,
+  which means we likely need to introduce the notion of unmanaged 
+  resources (which the operator watches but doesn't own) for the purpose of 
+  allowing references to those resources. This puts some additional burden on the customers mental model
+  as in order to know what an action will do on a resource they need to know if it's managed or unmanaged.
+
+### Implementation
+There are three key pillars required for implementing the proposal as discussed
+
+#### Identify resource relationships
+
+##### Resource references (for related resources)
+We must identify all resource references in the JSON schema and transform their type to `AugmentedKnownTypeReference`.
+There is no specific marker which means: "This field is a reference" - most are called `id` but that's not a guarantee.
+For example on the [VirtualMachineScaleSetIPConfigurationProperties](https://schema.management.azure.com/schemas/2019-07-01/Microsoft.Compute.json) 
+the `subnet` field is of custom type `ApiEntityReference`, which has an `id` field where you put the ARM ID for a subnet.
+This may require some manual work. One thing we can investigate doing long term
+is see if there's a way to get teams to annotate "links" in their Swagger somehow...
```

Comment:
At this level of nesting, I found using a bold sentence fragment at the start of a paragraph read better. 

For example:

> **For references to related resources** found in the JSON schema, we must transform their type to `AugmentedKnownTypeReference`.
> There is no specific marker which means: "This field is a reference" - most are called `id` but that's not a guarantee.
> ...
> **For references to owned/owning resources**, we ...

Reply:
Suggestion applied